node/manifest.ts

```diff
@@ -134,20 +140,32 @@ const pathToRegularExpression = (path: string) => {
   // for both `/foo` and `/foo/`.
   const normalizedSource = `^${regularExpression.source}\\/?$`
 
-  return new RegExp(normalizedSource)
+  return normalizedSource
 }
 
 const getRegularExpression = (declaration: Declaration) => {
   if ('pattern' in declaration) {
-    return new RegExp(declaration.pattern)
+    try {
+      return parsePattern(declaration.pattern)
+    } catch (error: unknown) {
+      throw new Error(
+        `Could not parse path declaration of function '${declaration.function}': ${(error as Error).message}`,
+      )
+    }
   }
 
   return pathToRegularExpression(declaration.path)
 }
 
 const getExcludedRegularExpression = (declaration: Declaration) => {
   if ('pattern' in declaration && declaration.excludedPattern) {
```

Comment:
```suggestion
  if ('excludedPattern' in declaration && declaration.excludedPattern) {
```

? Not really a change from this PR but seems wit was already wrong?

Reply:
Good catch! Fixed in https://github.com/netlify/edge-bundler/pull/298/commits/51d1023da226a871668faee74960267ed539fbee.